### PR TITLE
Enhancement of Time class

### PR DESCRIPTION
**Pull request by lablnet**
Fix for https://github.com/zestframework/Zest_Framework/issues/234

### CODE REVIEW

**Comment by lablnet** [comment]
What about `validation` Method? @peter279k

File: src/Time/Time.php [hunk 1]
```diff
@@ -21,6 +21,63 @@
...
+    /**
+     * Validate the time format.
+     *
+     * @param mixe $format Valiud Date/Time format.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+     public function setDateFormat($format)
```
Comment (peter279k) [inline src/Time/Time.php]: What kinds of types for `$format`? The `DateTime` class and `string`?

File: src/Time/Time.php [hunk 2]
```diff
@@ -82,8 +139,10 @@ public static function timestampToGmt($time = null)
         $time = $time ?? time();
         $dateTime = new \DateTime();
         $dateTime->setTimestamp($time)->modify('+2 hours');
+        $format = isset($this->dateFormat) ? $format : 'd/m/Y H:i:s';
```
Comment (peter279k) [inline src/Time/Time.php:146]: We can consider letting the `$this->dateFormat` be the default date time format string.

Then this line should be unnecessary.

File: src/Time/Time.php [hunk 3]
```diff
@@ -101,8 +160,9 @@ private static function timestampToDate($time = null)
         $time = self::formatTime($time);
         $dateTime = new \DateTime();
         $dateTime->setTimestamp($time);
+        $format = isset($this->dateFormat) ? $format : 'd-m-Y H:i:s';
```
Comment (peter279k) [inline src/Time/Time.php:167]: It has the same issue that I mention on the previous comment thread.

File: src/Time/Time.php (hunk 2 above)
Comment (lablnet) [reply src/Time/Time.php:146]: So it mean it should has default value as well, or user should need to set it?

File: src/Time/Time.php (hunk 1 above)
Comment (lablnet) [reply src/Time/Time.php]: Typo

Comment (lablnet) [reply src/Time/Time.php]: Yeah its Date Tine class

**Comment by peter279k** [comment]
BTW, the PR title should be `Enhancement`, not `Entrenchment`.

File: src/Time/Time.php (hunk 2 above)
Comment (peter279k) [reply src/Time/Time.php:146]: That's right.

File: src/Time/Time.php (hunk 1 above)
Comment (peter279k) [reply src/Time/Time.php]: @Lablnet, thanks.

But it seems that the `$format` type is `mixed` inside the this method comment annotation.

The `mixed` type comment annotation means this `$format` can allow two or more types.

Comment (lablnet) [reply src/Time/Time.php]: yes i will fix this but how/what about the validation process

Comment (peter279k) [reply src/Time/Time.php]: Could you illustrate the `validation` approach?

Comment (lablnet) [reply src/Time/Time.php]: idk you mentioned in issue

File: src/Time/Time.php [hunk 4]
```diff
@@ -21,6 +21,63 @@
...
+    /**
+     * Validate the time format.
+     *
+     * @param mixe $format Valiud Date/Time format.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+     public function validateDateFormat($format)
```
Comment (peter279k) [inline src/Time/Time.php:76]: Here is the [reference ](https://www.php.net/manual/en/function.checkdate.php#113205)about validating date format approach.

Comment (lablnet) [reply src/Time/Time.php:76]: oh, we need to validate Date with format or just format

is that what you propose

> Add the validateDateFormat method when setting setDateFormat

Comment (peter279k) [reply src/Time/Time.php:76]: I think we can validate date format, not just for the date format string.

**Pull request merged**